pkg/clusteragent/autoscaling/workload/scaler.go

```diff
@@ -0,0 +1,69 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package workload
+
+import (
+	"context"
+	"fmt"
+
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	scaleclient "k8s.io/client-go/scale"
+)
+
+type scaler interface {
+	get(ctx context.Context, namespace, name string, gvk schema.GroupVersionKind) (*autoscalingv1.Scale, schema.GroupResource, error)
+	update(ctx context.Context, gr schema.GroupResource, scale *autoscalingv1.Scale) (*autoscalingv1.Scale, error)
+}
+
+type scalerImpl struct {
+	restMapper  apimeta.RESTMapper
+	scaleGetter scaleclient.ScalesGetter
+}
+
+func newScaler(restMapper apimeta.RESTMapper, scaleGetter scaleclient.ScalesGetter) scaler {
+	return &scalerImpl{
+		restMapper:  restMapper,
+		scaleGetter: scaleGetter,
+	}
+}
+
+func (sg *scalerImpl) get(ctx context.Context, namespace, name string, gvk schema.GroupVersionKind) (*autoscalingv1.Scale, schema.GroupResource, error) {
+	mappings, err := sg.restMapper.RESTMappings(gvk.GroupKind())
```

Comment:
I think the `RESTMappings` function accepts the version as argument. Should we use it ?

Reply:
I am also curious about why we look at all the different resources. I would expect that Kubernetes has a single primary endpoint for a given group, version and kind for deployments / statefulsets...